api/maven-api-core/src/main/java/org/apache/maven/api/ExtensibleEnums.java

```diff
@@ -66,37 +107,73 @@ public String toString() {
         }
     }
 
+    /**
+     * Implementation of the PathScope interface.
+     */
     private static class DefaultPathScope extends DefaultExtensibleEnum implements PathScope {
         private final ProjectScope projectScope;
         private final Set<DependencyScope> dependencyScopes;
 
+        /**
```

Comment:
again not for this PR, but why is this all in one class? It feels like these should be split out to different classes